core/src/main/java/org/elasticsearch/index/shard/DocsStats.java

```diff
@@ -57,16 +62,38 @@ public long getDeleted() {
         return this.deleted;
     }
 
+    /**
+     * Returns the total size in bytes of all documents in this stats.
+     * This value may be more reliable than {@link StoreStats#getSizeInBytes()} in estimating the index size.
+     */
+    public long getTotalSizeInBytes() {
+        return totalSizeInBytes;
+    }
+
+    /**
+     * Returns the average size in bytes of all documents in this stats.
+     */
+    public long getAverageSizeInBytes() {
+        long totalDocs = count + deleted;
+        return totalDocs == 0 ? 0 : totalSizeInBytes / totalDocs;
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         count = in.readVLong();
         deleted = in.readVLong();
+        if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
```

Comment:
Also, this should be `V_6_1_0` since you will be backporting this to the 6.x branch

Reply:
@jpountz recommended to make this for v7, then change for the backport later.

Reply:
This is correct, it should be 7.0.0. Then when you backport set it to 6.1.0 in the 6.x branch and make sure that the BWC tests in master against 6.x pass (you might have to skip some of them). Then push a commit to master flipping the version to 6.1.0 and removing the skips.

Reply:
Ahh okay, I hadn't realized we were doing it the reverse way now :)

Reply:
I am not sure what you mean by the reverse way, these are part of the steps to have green CI on all branches every step of the way.